compiler/src/dmd/expressionsem.d

```diff
@@ -12796,6 +12796,59 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             return;
         }
 
+        // Inline the expression, if possible.
+        BinExp be = cast(BinExp)e;
+        if (be.e1.type.isScalar() && be.e2.isIntegerExp())
+        {
+            Expression one;
+            if (be.e1.type.isIntegral()) {
+                one = new IntegerExp(e.loc, 1, be.e1.type);
+            } else {
+                one = new RealExp(e.loc, CTFloat.one, be.e1.type);
+            }
+
+            const expo = cast(sinteger_t)be.e2.toInteger();
+            // Replace e1 ^^ -1 with 1 / e1
+            if (expo == -1)
+            {
+                Expression ex = new DivExp(exp.loc, one, be.e1);
+                ex = ex.optimize(WANTvalue);
```

Comment:
Is this constant folding here necessary? In theory that should be handled after expression semantic, but maybe you encountered a test failure?

Reply:
I think in an before moving the code to semantic analysis const expressions weren't folded when I inspected `-vcg-ast` but testing now, it const folds just fine.

This line is unnecessary.

Reply:
Addressed in the last commit, https://github.com/dlang/dmd/pull/21082/commits/72754cdebc284e7d87d8dd693d5b2d41a676fdf7.